lib/galaxy/tool_util/lint.py

```diff
@@ -89,59 +124,67 @@ def __init__(self, level, skip_types=None, object_name=None):
+            # was reset when called with a new lint_func, as workaround
+            # we safe the message list, apply the lint_func (which then
+            # adds to the message_list) and restore the message list
+            # at the end (+ append the new messages)
+            tmp_message_list = list(self.message_list)
+            self.message_list = []
 
         # call linter
         lint_func(lint_target, self)
-        # TODO: colorful emoji if in click CLI.
-        if self.error_messages:
-            status = "FAIL"
-        elif self.warn_messages:
-            status = "WARNING"
-        else:
-            status = "CHECK"
-
-        def print_linter_info():
-            if self.printed_linter_info:
-                return
-            self.printed_linter_info = True
-            print(f"Applying linter {name}... {status}")
-
-        for message in self.message_list:
-            if message.level != "error":
-                continue
-            self.found_errors = True
-            print_linter_info()
-            print(f"{message}")
-
-        if self.level != LEVEL_ERROR:
-            for message in self.message_list:
-                if message.level != "warning":
-                    continue
-                self.found_warns = True
-                print_linter_info()
-                print(f"{message}")
 
-        if self.level == LEVEL_ALL:
-            for message in self.message_list:
-                if message.level != "info":
-                    continue
-                print_linter_info()
-                print(f"{message}")
-            for message in self.message_list:
-                if message.level != "check":
-                    continue
-                print_linter_info()
+        if not silent:
+            # TODO: colorful emoji if in click CLI.
+            if self.error_messages:
+                status = "FAIL"
+            elif self.warn_messages:
+                status = "WARNING"
+            else:
+                status = "CHECK"
+
+            def print_linter_info(printed_linter_info):
+                if printed_linter_info:
+                    return True
+                print(f"Applying linter {name}... {status}")
+                return True
+
+            plf = False
+            for message in self.error_messages:
+                plf = print_linter_info(plf)
                 print(f"{message}")
 
+            if self.level != LEVEL_ERROR:
+                for message in self.warn_messages:
+                    plf = print_linter_info(plf)
+                    print(f"{message}")
+
+            if self.level == LEVEL_ALL:
+                for message in self.info_messages:
+                    plf = print_linter_info(plf)
+                    print(f"{message}")
+                for message in self.valid_messages:
+                    plf = print_linter_info(plf)
+                    print(f"{message}")
+            self.message_list = tmp_message_list + self.message_list
```

Comment:
I think `tmp_message_list` can be unbound here if we don't declare it as an empty list before line 144?

Reply:
Its both times `if not silent` .. so it should be fine. But if you prefer I can add the extra line.

Reply:
You are right, I guess there's no need for the extra line